config_verifier/src/index.ts

```diff
@@ -0,0 +1,520 @@
+                data = await configureSolana(rpc, managerAddress);
+                data['type'] = chain['networkType'];
+                configurationData[data['chainId']] = data; 
+            }
+            else {
+                console.log(`ERROR: Not supported networkType ${chain['networkType']}`);
+                return [null, `ERROR: Not supported networkType ${chain['networkType']}`];
+            }
+        }catch(e){
+            console.log(`Error: ${e}`);
+            console.log("Exiting...");
+            return [null, e];
+        }
+
+        // Checking to ensure that the wormhole chain id exists
+        if(!wormholeChainIds.includes(data['chainId'])){
+            console.log(`ERROR: Provided chainId is not supported on Wormhole - ${chain['chainId']}. This is the wormhole chain id not the actual chain id, found at https://docs.wormhole.com/wormhole/reference/constants.`)
+            return [null, `ERROR: Provided chainId is not supported on Wormhole - ${chain['chainId']}`];
+        }
+
+    }
+
+    return [configurationData, null];
+}
+
+async function configureSolana(rpc, managerAddress){
+    const managerData = {}; 
+
+    const SOL_CONNECTION = new web3.Connection(
+        rpc,
+        "confirmed"
+    );
+
+    // Devnet configuration....
+    const fromHexString = (hexString) => Uint8Array.from(hexString.match(/.{1,2}/g).map((byte) => parseInt(byte, 16)));
+    managerData['managerAddress'] = managerAddress;
+    const manager = new NTT(SOL_CONNECTION, {
+        nttId: base58.encode(fromHexString(managerAddress.substring(2))),
+        wormholeId: "3u8hJUVTA4jH1wYAyUur7FFZVQ8H635K3tSHHF4ssjQ5",
+      });
+
+
+    var to_public_key = new web3.PublicKey(
+        base58.encode(fromHexString(managerAddress.substring(2)))
+    );
+    // var accounts = await SOL_CONNECTION.getProgramAccounts(to_public_key, {filters: [{
+    //     memcmp: {
+    //         offset: 0,
+    //         bytes: base58.encode([231, 104, 182, 96, 168, 43, 216, 20])
+    //     }
+    // }]});
+
+    // console.log(accounts);
+    // //console.log(accounts.length)
+    // console.log(await manager.program.account.registeredTransceiver.fetch(accounts[0].pubkey));
+    const managerConfig = await manager.getConfig();
+
+    managerData['mode'] = (managerConfig['mode'].locking == undefined ? 1 : 0); // Convert to an integer for later use
+    managerData['token'] = managerConfig['tokenProgram']
+    managerData['threshold'] = managerConfig['threshold'].valueOf() // The 'threshold' value
+    managerData['count'] = managerConfig.enabledTransceivers; // Amount of active transceivers
+    managerData['chainId'] = 1;
+    
+    // https://spl.solana.com/token#example-creating-your-own-fungible-token
+    const mintInfo = await spl.getMint(
+        SOL_CONNECTION,
+        managerConfig['mint'],
+        'finalized',
+      );
+        
+    managerData['token'] = "0x" + bufferToHex(managerConfig['mint'].toBuffer()); 
```

Comment:
I think it's worth a comment here that the `0x` prefix is being added for convenience to allow comparing Solana/Eth with the same functions. 
It looks a little strange to see Solana addresses with a `0x` in front since these aren't used in the ecosystem (and in fact program addresses are not hex-encoded but use base58).